### PR TITLE
LogDensityProblemsAD + fix CI bug

### PR DESCRIPTION
**Pull request by willtebbutt**
- [x] make TemporalGPs integration testing actually run during CI (I forgot to do this in the last PR)
- [x] add extension to interface with LogDensityProblemsAD

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -41,6 +41,7 @@ include("front_matter.jl")
         end
         include("chain_rules_macro.jl")
     elseif test_group == "integration_testing/misc"
+        include(joinpath("integration_test", "logdensityproblemsad_interop.jl"))
```
Comment (yebai) [inline test/runtests.jl]:
```suggestion
        include(joinpath("integration_testing", "logdensityproblemsad_interop.jl"))
```

**Comment by willtebbutt** [comment]
@yebai please don't merge -- need to add ADTypes interop still.

**Comment by willtebbutt** [comment]
Yup, I just need to make sure that interop between it and LogDensityProblemsAD is done correctly

**Comment by willtebbutt** [comment]
Okay. @yebai I can't support ADTypes yet because it's held back by Turing -- the current version restricts it to 0.2. @torfjeld mentioned to me that this will be bumped at some point in the near future, so I'll revisit when that happens.

**Comment by devmotion** [comment]
Should this be moved to LogDensityProblemsAD? Generally, there's no clear rule regarding which package defines an extension but since all other AD backends are extensions in LogDensityProblemsAD that would seem a bit more consistent.

**Comment by willtebbutt** [comment]
I think for now it's more straightforward for me to have this code here so that I have direct control over it, and can trivially fix it when I make breaking changes to Tapir, but once things have stabilised a bit I agree that it would probably make sense to move it over to LogDensityProblemsAD

**Pull request merged**